tests/cpgrid/communicate_distributed_grid_with_lgrs_test.cpp

```diff
@@ -0,0 +1,297 @@
+        char** m_argv = boost::unit_test::framework::master_test_suite().argv;
+        Dune::MPIHelper::instance(m_argc, m_argv);
+        Opm::OpmLog::setupSimpleDefaultLogging();
+    }
+};
+
+BOOST_GLOBAL_FIXTURE(Fixture);
+
+void createTestGrid(Dune::CpGrid& grid)
+{
+    Opm::Parser parser;
+    const std::string deck_string = R"(
+RUNSPEC
+DIMENS
+  4 4 1 /
+GRID
+CARFIN
+-- NAME I1-I2 J1-J2 K1-K2 NX NY NZ
+'LGR1'  2  3  2  3  1  1  6  6  3/
+ENDFIN
+DX
+  16*1000 /
+DY
+	16*1000 /
+DZ
+	16*20 /
+TOPS
+	16*8325 /
+ACTNUM
+        16*1 /
+PORO
+  16*0.15 /
+PERMX
+  16*1 /
+COPY
+  PERMX PERMZ /
+  PERMX PERMY /
+/
+EDIT
+OIL
+GAS
+TITLE
+The title
+START
+16 JUN 1988 /
+PROPS
+REGIONS
+SOLUTION
+SCHEDULE
+)";
+
+    const auto deck = parser.parseString(deck_string);
+    Opm::EclipseState ecl_state(deck);
+    Opm::EclipseGrid eclipse_grid = ecl_state.getInputGrid();
+    grid.processEclipseFormat(&eclipse_grid, &ecl_state, false, false, false);
+}
+
+void checkChildGlobalIdsTest(const Dune::CpGrid& grid)
+{
+    // first_child_ids [ parent cell global id ] =  first child cell global id from undistributed view.
+    const std::unordered_map<int,int> first_child_ids = { {5, 66}, {6, 93}, {9, 120}, {10, 147} };
+    // parent cell with global id equal to 5 has children cell ids 66, ..., 92 (= 66 + 26).
+    // parent cell with global id equal to 6 has children cell ids 93, ..., 119 (= 119 + 26).
+    // parent cell with global id equal to 9 has children cell ids 120, ..., 146 (= 120 + 26).
+    // parent cell with global id equal to 10 has children cell ids 147, ..., 173 (= 147 + 26).
+
+    for (const auto& element : elements(grid.leafGridView())) {
+        if (!element.hasFather()) {
+            continue;
+        }
+        const auto& parent_globalId = grid.globalIdSet().id(element.father());
+        const auto& idx_in_parent = element.getIdxInParentCell();
+
+        const auto& expected_elem_globalId = first_child_ids.at(parent_globalId) + idx_in_parent;
+        const auto& actual_elem_globalId = grid.globalIdSet().id(element);
+        BOOST_CHECK_EQUAL( expected_elem_globalId, actual_elem_globalId);
+    }
+}
+
+void checkCommunicationWorks(const Dune::CpGrid& grid)
+{
+    // This pretends to do something on the lines
+    // of fvbasadiscretization finishInit()
+
+    double gridTotalVolume = 0.0;
+    std::vector<double> dofTotalVolume{};
+
+    const auto& gridView = grid.leafGridView();
+    // initialize the volume of the finite volumes to zero
+    const std::size_t numDof = gridView.size(0);
+    dofTotalVolume.resize(numDof);
+    std::fill(dofTotalVolume.begin(), dofTotalVolume.end(), 0.0);
+
+    for (const auto& element : Dune::elements(gridView)) {
+        // ignore everything which is not in the interior if the
+        // current process' piece of the grid
+        if (element.partitionType() != Dune::InteriorEntity) {
+            continue;
+        }
```

Comment:
Could be simplified to 
```suggestion
    for (const auto& element : Dune::elements(gridView, Dune::Partitions::interior)) {
        // ignore everything which is not in the interior if the
        // current process' piece of the grid
```